ambry-api/src/main/java/com/github/ambry/rest/RestUtils.java

```diff
@@ -1011,6 +1033,20 @@ private static void encodeMetadataValueAndSetHeader(String headerName, String he
     }
   }
 
+  /**
+   * Encode a map of {@link String}s as http header value.
+   * @param map {@link Map} of {@link String}s to encode.
+   * @return encoded http header value.
+   */
+  static String encodeKVHeaderValue(Map<String, String> map) {
+    Objects.requireNonNull(map);
+    String value = "";
+    for(Map.Entry<String, String> entry : map.entrySet()) {
+      value += entry.getKey() + "=" + entry.getValue() + "; ";
```

Comment:
Semicolon is non-standard for key-value string format,  please use comma (and remove trailing one).
Eg.`"READ_CAPACITY_UNIT=1.9, STORAGE_IN_GB=3.0" `
I think there's a utility to print a map out this way but need to look it up.
That way, the header value can be parsed as a Property map.

Reply:
Would you mind adding a public method to reverse parse the header value back to a Map<String,Double>?  That way I can just call it and not worry about the format?

Reply:
done.